liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/BasicSupport.java

```diff
@@ -458,4 +472,26 @@ private InputStream getZipEntry(File zip, String entry) throws IOException {
         }
         return null;
     }
+
+    // Read WLP_OUTPUT_DIR from server.env. Return null if server.env doesn't
+    // exist or variable is not in server.env
+    private String getWlpOutputDir() throws IOException {
+        Properties envvars = new Properties();
+        
+        File serverEnvFile = new File(installDirectory, "etc/server.env");
+        if (serverEnvFile.exists()) {
+            envvars.load(new FileInputStream(serverEnvFile));
+        }
+        
+        if (serverEnv.exists()) {
```

Comment:
Shouldn't this `if` be an `else` to the `if` below since the `configDirectory` takes priority and we would not merge the properties in `configDirectory` and `serverEnv` in that case?  They would get one or the other.  I see why you merged the `etc/server.env` variables from above.

```
        serverEnvFile = new File(configDirectory, "server.env");
        if (configDirectory != null && serverEnvFile.exists()) {
            envvars.load(new FileInputStream(serverEnvFile));
        } else if (serverEnv.exists()) {
            envvars.load(new FileInputStream(serverEnv));
        }
```

Reply:
I will use your suggestion which will handle/merge the other envvar properly.